src/python/pants/option/config.py

```diff
@@ -281,22 +265,10 @@ def _stringify_val_without_interpolation(self, raw_value: _TomlValue) -> str:
             interpolate=False,
         )
 
-    @property
-    def sections(self) -> list[str]:
-        return [scope for scope in self.values if scope != "DEFAULT"]
-
-    def has_section(self, section: str) -> bool:
-        return section in self.values
-
-    def has_option(self, section: str, option: str) -> bool:
-        if not self.has_section(section):
-            return False
-        return option in self.values[section] or option in self.defaults
-
-    def get_value(self, section: str, option: str) -> str:
-        section_values = self.values.get(section)
+    def get_value(self, section: str, option: str) -> str | None:
+        section_values = self.section_to_values.get(section)
         if section_values is None:
-            raise NoSectionError(section)
+            return None
```

Comment:
We no longer need NoSectionError and NoOptionError. They were used for clunky control flow across the abstraction boundaries, but we never actually cared about the distinction between them. Using None is fine.

Reply:
Agreed. I think this was a leftover of supporting both pants.ini and pants.toml at the same time.